build_tools/cmake/iree_copts.cmake

```diff
@@ -284,6 +284,23 @@ if(IREE_DEV_MODE)
   )
 endif()
 
+#-------------------------------------------------------------------------------
+# Enable frame pointers in IREE_DEV_MODE and in RelWithDebInfo build type ---
+# these are conditions under which the instrumentation/debugging/profiling
+# benefits of frame pointers outweigh the cost.
+#
+# `perf record -g` defaults to relying on frame pointers. There is a non-default
+# option `perf record --call-graph=dwarf`, to rely on debug info instead. It
+# produces inferior results and has a discoverability issue anyway.
+#-------------------------------------------------------------------------------
+
+if (IREE_DEV_MODE OR (CMAKE_BUILD_TYPE STREQUAL "RelWithDebInfo"))
```

Comment:
Build types are case insensitive, so this check should use something other than `STREQUAL` (e.g. convert to uppercase then check)

https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html
https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#build-configurations
https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#case-sensitivity
https://gitlab.kitware.com/cmake/cmake/-/issues/22591

Reply:
Thanks for the tip. Done, copied the TOUPPER line from `function(iree_fix_ndebug)` in our root CMakeLists.txt.